### PR TITLE
Fix/plugins workbox webpack plugin

### LINKED ISSUE #2376: webpackCompilationPlugins doesn't seem to work
**Library Affected**:
workbox-webpack-plugin@5.0.0

**Browser & Platform**:
all browsers

**Issue or Feature Request Description**:

`webpackCompilationPlugins` doesn't apply `webpack.DefinePlugin` and `webpack.EnvironmentPlugin` to the output bundle.   

The following does not replace `__REVISION__` in sw.js:

```js
new InjectManifest({
  swSrc: './src/sw/index.js',
  swDest: 'sw.js',
  webpackCompilationPlugins: [
    new webpack.DefinePlugin({
      __REVISION__: JSON.stringify(process.env.COMMIT_SHA1)
    })
  ]
})
```

### PR DESCRIPTION
**Pull request by prateekbh**
Fixes #2376

*Description of what's changed/fixed.*
The application of plugins is now a separate step, making sure that childCompiler is aware of the necessary changes before the plugins are applied

### CODE REVIEW

File: test/workbox-webpack-plugin/static/sw-src-define-plugin.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/*
+  Copyright 2018 Google LLC
```
Comment (philipwalton) [inline test/workbox-webpack-plugin/static/sw-src-define-plugin.js]: Nit: can you change this to 2020? It also helps to know at a glance when a file was added to the repo.

File: test/workbox-webpack-plugin/node/inject-manifest.js [hunk 2]
```diff
@@ -1239,6 +1239,57 @@ describe(`[workbox-webpack-plugin] InjectManifest (End to End)`, function() {
...
+        ],
+      };
+
+      const compiler = webpack(config);
+      compiler.run(async (webpackError, stats) => {
+        const swFile = upath.join(outputDir, 'service-worker.js');
+        try {
+          webpackBuildCheck(webpackError, stats);
+
+          const files = await globby(outputDir);
```
Comment (jeffposnick) [inline test/workbox-webpack-plugin/node/inject-manifest.js]: This is needed to get the tests working on Windows:

```suggestion
          const files = await globby('**', {cwd: outputDir});
```

File: test/workbox-webpack-plugin/static/sw-src-define-plugin.js (hunk 1 above)
Comment (prateekbh) [reply test/workbox-webpack-plugin/static/sw-src-define-plugin.js]: done

File: test/workbox-webpack-plugin/node/inject-manifest.js (hunk 2 above)
Comment (prateekbh) [reply test/workbox-webpack-plugin/node/inject-manifest.js]: done

**Pull request merged**